src/main/java/org/minions/devfund/carlosorellana/Squarelotron.java

```diff
@@ -0,0 +1,140 @@
+package org.minions.devfund.carlosorellana;
+
+/**
+ * The goal is execute the Squarelotron methods.
+ *
+ * @author Carlos Orellana.
+ * @since 07/17/2018.
+ */
+public class Squarelotron {
+
+    private int[][] squarelotron;
+    private int size;
+
+    /**
+     * Constructor Sqarelotron.
+     *
+     * @param n - Size of squarelotron array.
+     */
+    public Squarelotron(final int n) {
+        this.size = n;
+        this.squarelotron = new int[size][size];
+
+        for (int i = 0; i < size; i++) {
+            for (int j = 0; j < size; j++) {
+                squarelotron[i][j] = size * i + j + 1;
+            }
+        }
+    }
+
+    /**
+     * Builds new array in base a Squarelotron.
+     *
+     * @return New array with Squarelotron values.
+     */
+    public int[] buildArray() {
+        int[] newArray = new int[this.size * this.size];
+
+        for (int x = 0; x < this.size; x++) {
+
+            for (int y = 0; y < this.size; y++) {
+                newArray[this.size * x + y] = squarelotron[x][y];
+            }
+        }
+
+        return newArray;
+    }
+
+    /**
+     * Finds the Maximum Rings number.
+     *
+     * @return Numbers maximum of rings.
+     */
+    public int findMaxRings() {
+        return (this.size % 2 == 0)
+                ? size / 2
+                : size / 2 + 1;
+    }
+    /**
+      * Modifies A Upside-Down Flip of the outer ring of Squarelotron.
+      *
+      * @param ring - Number of ring.
+      * @return Instance new Squarelotron.
+      */
+    public Squarelotron upsideDownFlip(int ring) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 23 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)